components/lib/datatable/datatable.d.ts

```diff
@@ -1558,6 +1585,71 @@ export interface DataTableProps<TValue extends DataTableValueArray> extends Omit
     unstyled?: boolean;
 }
 
+interface DataTablePropsSingle<TValue extends DataTableValueArray> extends DataTableBaseProps<TValue> {
+    /**
+     * Whether to cell selection is enabled or not.
+     * @defaultValue false
+     */
+    cellSelection?: false | undefined;
+    /**
+     * Specifies the selection mode, valid values are "single", "multiple", "radiobutton" and "checkbox".
+     */
+    selectionMode?: 'single' | 'radiobutton' | undefined;
+    /**
+     * Selected row in single mode or an array of values in multiple mode.
```

Comment:
The comments need to be updated if we have a different type of single and multiple now